src/javascript/crypto/e2e/vrf/extra25519.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * @license
+ * Copyright 2015 Yahoo Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Helper functions for the Yahoo E2E vrf.verify() implementation
+ * This is ported from https://github.com/yahoo/coname/blob/master/vrf/vrf.go
+ */
+
+goog.provide('e2e.vrf.extra25519');
+
+goog.require('goog.asserts');
+goog.require('goog.crypt');
+goog.require('e2e.BigNum');
+goog.require('e2e.ecc.constant');
+goog.require('e2e.ecc.DomainParam');
+goog.require('e2e.ecc.PrimeCurve');
+
+goog.require('e2e.vrf.sha3Shake256');
```

Comment:
the changes are prepared at https://gist.github.com/adon-at-work/ca254eed709ecf322d47
will commit it to this PR at last, as to avoid collapsing all the comments due to "outdated diff".